ouroboros-consensus/src/Ouroboros/Consensus/ChainSyncClient.hs

```diff
@@ -376,9 +526,6 @@ chainSyncClient mkPipelineDecision0
       -- the transaction, because we might have to retry the transaction if the
       -- ledger state is too far behind the upstream peer (see below).
       curLedger <- ledgerState <$> getCurrentLedger
```

Comment:
Let's document that we don't need this to be consistent with our view of our chain: we use it _only_ to validate _their_ header.

Reply:
(Which is true even for the special case below.)